stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -108,6 +108,8 @@ openblas_get_config() = strip(unsafe_string(ccall((@blasfunc(openblas_get_config
     set_num_threads(n)
 
 Set the number of threads the BLAS library should use.
+If `n <= 0` then the number of threads will not be changed.
+If `n`  is greater than number of CPU threads, then the nuumber of BLAS threads wiol be set to the number of CPU threads.
```

Comment:
```suggestion
If `n`  is greater than number of CPU threads, then the nuumber of BLAS threads will be set to the number of CPU threads.
```

Reply:
Is this hardware threads?  I think it actually includes hyper threads, which kinda sucks.  Does it also happen with MKL and other BLAS's?  Should we just do this in Julia (clamping to hardware threads) before punting to the library?

Reply:
I assume it is logical threads, (so yes, including hyperthreading.)

My understanding is that locking down to only physical threads isn't that good except on dedicated machines that only run 1 process at a time.
And such machines just disable hyperthreading in the BIOS.

I feel like the authors of OpenBLAS would have already put the thought into it.
And if there is a better thing to do, then we should push it upstream as a PR to OpenBLAS.


I still need to go digging in the MKL docs.